framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -195,6 +194,11 @@ public AnnotatedTypeMirror visitMemberSelect(MemberSelectTree node,
             if (t instanceof AnnotatedDeclaredType || t instanceof AnnotatedArrayType || t instanceof AnnotatedTypeVariable) {
                 return AnnotatedTypes.asMemberOf(f.types, f, t, elt).asUse();
             }
+        } else {
+            if(node.getIdentifier().contentEquals("class")){
+                // Handle class literals for primitive types (as in int.class)
+                return f.getAnnotatedType(elt);
```

Comment:
The main change is that in "primitive.class" we now apply the implicits for the primitive type to the result, instead of just taking the types from the element (which is what fromElement did).
Why do we want to apply the implicits here instead of in ImplicitsTreeAnnotator?
Can you maybe improve the class documentation and document what the input and output properties are?